src/mrinufft/operators/interfaces/gpunufft.py

```diff
@@ -0,0 +1,275 @@
+    shape: tuple of int
+        shape of the image
+    operator: The NUFFTOp object
+        to carry out operation
+    n_coils: int default 1
+            Number of coils used to acquire the signal in case of multiarray
+            receiver coils acquisition. If n_coils > 1, please organize data as
+            n_coils X data_per_coil
+    """
+
+    def __init__(
+        self,
+        samples,
+        shape,
+        n_coils=1,
+        density_comp=None,
+        kernel_width=3,
+        sector_width=8,
+        osf=2,
+        balance_workload=True,
+        smaps=None,
+    ):
+        """Initilize the 'NUFFT' class.
+
+        Parameters
+        ----------
+        samples: np.ndarray
+            the kspace sample locations in the Fourier domain,
+            normalized between -0.5 and 0.5
+        shape: tuple of int
+            shape of the image
+        n_coils: int
+            Number of coils used to acquire the signal in case of multiarray
+            receiver coils acquisition
+        density_comp: np.ndarray default None.
+            k-space weighting, density compensation, if not specified
+            equal weightage is given.
+        kernel_width: int default 3
+            interpolation kernel width (usually 3 to 7)
+        sector_width: int default 8
+            sector width to use
+        osf: int default 2
+            oversampling factor (usually between 1 and 2)
+        balance_workload: bool default True
+            whether the workloads need to be balanced
+        smaps: np.ndarray default None
+            Holds the sensitivity maps for SENSE reconstruction
+        """
+        if GPUNUFFT_AVAILABLE is False:
+            raise ValueError(
+                "gpuNUFFT library is not installed, " "please refer to README"
+            )
+        if (n_coils < 1) or not isinstance(n_coils, int):
+            raise ValueError("The number of coils should be an integer >= 1")
+        self.n_coils = n_coils
+        self.shape = shape
+        self.samples = proper_trajectory(samples, normalize="unit")
+        if density_comp is None:
+            density_comp = np.ones(samples.shape[0])
+        if smaps is None:
+            self.uses_sense = False
+        else:
+            smaps = np.asarray(
+                [np.reshape(smap_ch.T, smap_ch.size) for smap_ch in smaps]
+            ).T
+            self.uses_sense = True
+        self.operator = NUFFTOp(
+            np.reshape(samples, samples.shape[::-1], order="F"),
+            shape,
+            n_coils,
+            smaps,
+            density_comp,
+            kernel_width,
+            sector_width,
+            osf,
+            balance_workload,
+        )
+
+    def op(self, image, interpolate_data=False):
+        """Compute the masked non-cartesian Fourier transform.
+
+        Parameters
+        ----------
+        image: np.ndarray
+            input array with the same shape as shape.
```

Comment:
Maybe state that the second shape is the one belonging to the object (like "self.shape" or something similar)